packages/react/src/index.ts

```diff
@@ -19,64 +20,91 @@ function useForceUpdate() {
   return useRef(() => update(v => v + 1)).current
 }
 
-function useUnsubscribe(ref: MutableRefObject<Function>) {
-  useEffect(
-    () => () => {
-      ref.current()
-    },
-    [ref],
-  )
-}
+const lifeCycleStatus = {
+  initActual: 'initActual',
+  initNotActual: 'initNotActual',
+  mounted: 'mounted',
+} as const
 
 /**
- * @param atom target atom for subscription
+ * @param atom target atom for subscription or callback for it creation
  * @param isUpdatesNotNeeded usefully for mount lazy atoms, but not subscribe to it updates
  * @returns atom value
  */
-export function useAtom<T>(
-  atom: Atom<T> | (() => Atom<T>),
-  isUpdatesNotNeeded = false,
-): T {
-  const atomRef = useRef<Atom<T>>()
-  const stateRef = useRef<T>()
-  const unsubscribeRef = useRef<Function>()
-  const isMountRef = useRef(false)
+export function useAtom<T>(atom: Atom<T>): T
+export function useAtom<TI, TO = TI>(
+  atom: Atom<TI>,
+  mapper: (atomValue: TI) => TO,
+  deps: any[],
+): TO
+export function useAtom<TI, TO = TI>(
+  atom: Atom<TI>,
+  mapper: (atomValue: TI) => TO = (atomValue: TI) =>
+    (atomValue as unknown) as TO,
+  deps: any[] = [],
```

Comment:
`deps?: any[] = [],`

Reply:
An argument with default value is optional (by typings) as is